src/portal/lib/src/config/system/system-settings.component.html

```diff
@@ -56,22 +63,94 @@
+                                </clr-textarea-container>
+                            </div>
+                            <div>
+                                <button [disabled]="isDisabled()" (click)="addToSystemWhitelist()"
+                                        class="btn btn-link">{{'CVE_WHITELIST.ADD'|translate}}</button>
+                            </div>
+                        </div>
+                        <ul class="whitelist-window">
+                            <li *ngIf="systemWhitelist?.items?.length<1"
+                                class="none">{{'CVE_WHITELIST.NONE'|translate}}</li>
+                            <li *ngFor="let item of systemWhitelist?.items;let i = index;">{{item.cve_id}}
+                                <clr-icon (click)="deleteItem(i)" class="float-lg-right margin-top-4"
+                                          shape="times-circle"></clr-icon>
+                            </li>
+                        </ul>
+                    </div>
+                    <div class="clr-col padding-top-8">
+                        <div class="form-group padding-left-80">
+                            <label for="expires">{{'CVE_WHITELIST.EXPIRES_AT'|translate}}</label>
+                            <div class="underline">
+                                <input #dateInput placeholder="{{'CVE_WHITELIST.NEVER_EXPIRES'|translate}}" readonly
+                                       type="date" [(clrDate)]="expiresDate" newFormLayout="true">
+                            </div>
+                        </div>
+                        <div class="form-group padding-left-80">
+                            <clr-checkbox-wrapper>
+                                <input [checked]="neverExpires" [(ngModel)]="neverExpires" type="checkbox" clrCheckbox
+                                       name="neverExpires" id="neverExpires"/>
+                                <label>
+                                    {{'CVE_WHITELIST.NEVER_EXPIRES'|translate}}
+                                </label>
+                            </clr-checkbox-wrapper>
+                        </div>
+                    </div>
+                </div>
+
+            </div>
+        </div>
     </section>
 </form>
 <div>
-    <button type="button" id="config_system_save" class="btn btn-primary" (click)="save()" [disabled]="!isValid() || !hasChanges()">{{'BUTTON.SAVE'
+    <button type="button" id="config_system_save" class="btn btn-primary" (click)="save()"
+            [disabled]="(!isValid() || !hasChanges()) && (!hasWhitelistChanged) || inProgress">{{'BUTTON.SAVE'
         | translate}}</button>
-    <button type="button" class="btn btn-outline" (click)="cancel()" [disabled]="!isValid() || !hasChanges()">{{'BUTTON.CANCEL'
+    <button type="button" class="btn btn-outline" (click)="cancel()"
+            [disabled]="(!isValid() || !hasChanges()) && (!hasWhitelistChanged) || inProgress">{{'BUTTON.CANCEL'
```

Comment:
(!isValid() || !hasChanges()) && (!hasWhitelistChanged) || inProgress;
the logic is wrong